os/binfmt/elf.c

```diff
@@ -236,6 +236,11 @@ static int elf_loadbinary(FAR struct binary_s *binp)
 	loadinfo.offset = binp->offset;
 	loadinfo.filelen = binp->filelen;
 	loadinfo.compression_type = binp->compression_type;
```

Comment:
Can we merge those lines?
```
loadinfo.compression_type = binp->compression_type;
#ifdef CONFIG_COMPRESSED_BINARY
	if (binp->islibrary) {
		loadinfo.compression_type = CONFIG_COMPRESSION_TYPE;
	}
#endif
```
Because we can always set the compression type with `CONFIG_COMPRESSION_TYPE`.
```
#ifdef CONFIG_COMPRESSED_BINARY
    loadinfo.compression_type = CONFIG_COMPRESSION_TYPE;
#else
    loadinfo.compression_type = binp->compression_type;
```

Reply:
True.. Actually we dont need to set the compression type at all. We will always use CONFIG_COMPRESSION_TYPE if compression is enabled. So shall I remove the whole code related to setting compression type in code and binary header?

Reply:
I agree to remove redundant codes about setting compresstion type.
But even in binary header? Why?
I think we should check the header that binary is compressed or not. Isn't it?

Reply:
If `CONFIG_COMPRESSED_BINARY` is enabled, then we will always compress the binary using `CONFIG_COMPRESSION_TYPE`. It cannot be anything different. Then, why do we need to add this information in header?

Reply:
Umm.. When kernel build, enable the `CONFIG_COMPRESSED_BINARY`.
Then I think binary manager can support both compressed and non-compressed binary simultaneously.
If we can distinguish both type, then it is much better to manage the binaries against memory constraint.
How do you think?

Reply:
In that case, it is ok. But currently in the code, if `CONFIG_COMPRESSED_BINARY` is enabled, then we always create compressed binary. Also, if this is the case, then it is possible that common binary also can be either compressed or uncompressed. But, there is no way to distinguish it.

Reply:
As discussed through messenger, let's support same compression way for all binaries.
Please remove unnecessary codes which related to it.
Thanks :)

Reply:
Done